devel/git-remote-hg/Portfile

```diff
@@ -16,23 +18,27 @@ checksums           rmd160  433a2de35530aa3374f9001650c11e382530643f \
                     sha256  30fd7c3bbea5e0a98503378ca040fa6faa89fae8e22a06c0ac172388fb6ca92d \
                     size    65780
 
-depends_build       port:asciidoc
-depends_lib         port:mercurial \
-                    port:python38
+python.default_version 39
+
+depends_build       port:asciidoc \
+                    port:py${python.version}-setuptools
+depends_lib-append  port:mercurial
 
 post-patch {
-    reinplace "1 s|/usr/bin/env python.*|${prefix}/bin/python3.8|g" \
+    reinplace "1 s|/usr/bin/env python.*|${python.bin}|g" \
         ${worksrcpath}/git-remote-hg
 
     reinplace -E "s|^(prefix\[\[:space:\]\].*.?=).*|\\1 ${prefix}|g" \
         ${worksrcpath}/Makefile
 }
 
-use_configure       no
-
-build.type          gnu
+post-build {
```

Comment:
Is this here because the former declarations no longer work due to `python` PortGroup?

Reply:
Well, it's rather that they weren't needed; using the Python port group ensures that the Python modules are correctly installed, and this ensures that we still get the documentation. To be honest, I don't remember whether it didn't actually work; if you'd prefer to keep the previous style, I can look into it?

Reply:
This is a small detail and I don't want to add roadblocks to merge this PR, however if the "declarative style" works I think we should favour that because that's the idiomatic way and it leads to more maintainable ports.

Reply:
My general goal was to make it easier to keep the Python version in sync with Mercurial by using the Python port group. I _think_ the issue was that using that group with another build system is tricky, and it's much easier to just build the documentation separately.

Reply:
OK. If you tried and it didn't work, or it required too much fiddling, then it is fine as it is. I just wanted to understand the reasoning for the change.